### DISCUSSION TITLE
spin-spin correlation calculations by pyscf

### DISCUSSION THREAD

**Discussion by liurui191** [pyscf/pyscf#2642] [Q&A] [answered]
Hello, we plan to use pyscf to perform CASSCF single point calculations on various structures on the potential energy surface of chemical reactions. We have already selected the active space (10o, 10e), which is in order to obtain the spin-spin correlation functions between different radicals. According to other researchers, pyscf has implemented this functionality well based on DMRG. However, after carefully reading the pyscf documentation, we did not find specific details about computing spin-spin correlation functions. The input file provided is as follows, and we hope to receive support for calculating spin-spin correlation functions. Thank you very much for your help! Once again, thank you for your assistance!

from pyscf import gto, scf, lo

mol = gto.Mole()
mol.atom = 'H 0 0 0; F 0 0 1.1'
mol.basis = 'ccpvdz'
mol.build()

mf = scf.RHF(mol).run()

lo.orth_ao(mol, mf.mo_coeff)
dm = mf.make_rdm1() 

sscf = scf.hf.analyze(mol, mf.mo_coeff, dm)
spin_spin_corr = sscf.calculate_spin_spin_correlation()

print(spin_spin_corr)

**Comment by yangjunjie0320** [comment, marked as answer]
Hi, spin-spin correlation function is part of the spin-orbital 2-body density matrix. For CASSCF SSCF, you can refer to [this](https://github.com/pyscf/pyscf/issues/357) and the [related discussion from block2](https://github.com/block-hczhai/block2-preview/issues/114)
